src/crimson/osd/osd.cc

```diff
@@ -625,7 +613,17 @@ seastar::future<> OSD::shutdown()
   // TODO
   superblock.mounted = boot_epoch;
   superblock.clean_thru = osdmap->get_epoch();
```

Comment:
move messenger's shutdown to osd::shutdown(), then we can explicitly call it before engine_exit().